tests/validation/tests/v3_api/test_registry.py

```diff
@@ -115,8 +153,240 @@ def test_edit_registry_all_namespace():
+        p_client.delete(workload_invalid)
+        p_client.delete(workload_new)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_edit_registry_single_namespace(role):
+    """
+    Runs only if RANCHER_TEST_RBAC is True in env. variable.
+    Creates registry with invalid credential for single namespace,
+    deploys workload with invalid registry and validate the workload
+    is not up.
+    Update the registry with correct credential and validates workload.
+    @param role: User role in rancher eg. project owner, project member etc
+    """
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, token)
+
+    name = random_test_name("registry")
+    # Create registry with invalid username and password
+    registries = {REGISTRY: {"username": "testabc",
+                             "password": "abcdef"}}
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+        with pytest.raises(ApiError) as e:
+            registry = p_client.create_namespacedDockerCredential(
+                registries=registries, name=name,
+                namespaceId=ns.id)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'PermissionDenied'
+    else:
+        registry = p_client.create_namespacedDockerCredential(
+            registries=registries, name=name,
+            namespaceId=ns.id)
+
+        workload_invalid = \
+            create_validate_workload_with_invalid_registry(p_client, ns)
+        p_client.delete(workload_invalid)
+
+        # Update registry with valid username and password
+        new_registries = {REGISTRY: {"username": REGISTRY_USER_NAME,
+                                     "password": REGISTRY_PASSWORD}}
+        p_client.update(registry, name=registry.name,
+                        namespaceId=ns['name'],
+                        registries=new_registries)
+
+        # Validate workload after registry update
+        workload = create_validate_workload(p_client, ns)
+        delete_registry(p_client, registry, ns)
+        p_client.delete(workload)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_edit_registry_all_namespace(role):
+    """
+    Runs only if RANCHER_TEST_RBAC is True in env. variable.
+    Creates registry with invalid credential scoped for all namespace,
+    deploys workload with invalid registry and validate the workload
+    is not up.
+    Update the registry with correct credential and validates workload.
+    @param role: User role in rancher eg. project owner, project member etc
+    @param role:
+    """
+    c_client = namespace["c_client"]
+    cluster = namespace["cluster"]
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    p_client = get_project_client_for_token(project, token)
+    ns = rbac_get_namespace()
+
+    name = random_test_name("registry")
+
```

Comment:
Create registry as cluster owner and the check for edit as different roles.

Reply:
Done